### ISSUE TITLE
Local value never used false positive

### ISSUE THREAD

**Issue by auduchinok** [intellij-rust/intellij-rust#6994]
The `match` statement expression seems to be the culprit.

```rust
use std::io;
use rand::Rng;
use std::cmp::Ordering;

fn main() {
    println!("Guess the number!");
    println!("Please input your guess.");

    let secret_number = rand::thread_rng().gen_range(1..101);

    loop {
        let mut guess = String::new();
        io::stdin().read_line(&mut guess).expect("Failed to read line.");

        let guess: u32 =
            match guess.trim().parse() {
                
            };

        match guess.cmp(&secret_number) {
            Ordering::Less => println!("Too small!"),
            Ordering::Greater => println!("Too big!"),
            Ordering::Equal => {
                println!("You win!");
                break;
            }
        }
    }
}
```

<img width="626" alt="Screenshot 2021-03-19 at 19 03 38" src="https://user-images.githubusercontent.com/3923587/111809232-d22d3b00-88e5-11eb-8ea9-547c7e1a5ef7.png">

## Environment

* **IntelliJ Rust plugin version:** 0.3.143.3737-211
* **Rust toolchain version:** 1.50.0 aarch64-apple-darwin
* **IDE name and version:** CLion 2021.1 (CL-211.6432.9)
* **Operating system:** macOS 11.2.2
* **Macro expansion engine:** new
* **Name resolution engine:** new

**Comment by ortem** [comment]
Fixed in https://github.com/intellij-rust/intellij-rust/pull/7007

**Comment by auduchinok** [comment]
I'm using `0.3.145.3797-211`, and now there's a different false positive on the same code:
<img width="701" alt="Screenshot 2021-04-25 at 17 18 15" src="https://user-images.githubusercontent.com/3923587/115997077-6d52b800-a5ea-11eb-8ad7-303dc8d777bb.png">

@ortem Should I file another issue or reopen this one?
